app/assets/javascripts/discourse/app/components/site-header.js

```diff
@@ -484,17 +495,18 @@ export default SiteHeaderComponent.extend({
               "--header-top",
               `${headerTop}px`
             );
+            this.updateHeaderOffset();
           }
         }
       });
 
-      this._resizeObserver.observe(headerWrap);
+      this._resizeObserver.observe(this.headerWrap);
     }
   },
 
   willDestroyElement() {
     this._super(...arguments);
-
+    window.removeEventListener("scroll", () => this.updateHeaderOffset());
```

Comment:
Passing an anonymous function here won't work, because it technically has a different identity to the one passed to `addEventListener`. We need to `@bind` `updateHeaderOffset`, and then do something like


```
removeEventListener("scroll", this.updateHeaderOffset)
```

Reply:
I pushed a commit to fix this